Mathlib/Combinatorics/SimpleGraph/Walk.lean

```diff
@@ -1032,13 +1071,19 @@ theorem mem_support_iff_exists_getVert {u v w : V} {p : G.Walk v w} :
         rw [@nil_iff_length_eq]
         have : 1 ≤ p.length := by omega
         exact Nat.not_eq_zero_of_lt this
-      rw [← tail_support_eq_support_tail _ hnp]
+
+      rw [← support_tail_of_not_nil _ hnp]
       rw [mem_support_iff_exists_getVert]
       use n - 1
-      simp only [Nat.sub_le_iff_le_add, length_tail_add_one, getVert_tail]
+      simp only [Nat.sub_le_iff_le_add]
+      rw [getVert_tail _ hnp, length_tail_add_one hnp]
       have : n - 1 + 1 = n := by omega
       rwa [this]
```

Comment:
Can't omega just do it? 
```suggestion
      omega
```

Reply:
Nope, doesn't look like omega gets this in one go